src/Build/BackEnd/Components/SdkResolution/NuGet/RestoreRunnerEx.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.ProjectModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using ILogger = NuGet.Common.ILogger;
+
+namespace Microsoft.Build.BackEnd.SdkResolution.NuGet
+{
+    /// <summary>
+    /// An extension of the NuGet.Commands.RestoreRunner class that contains APIs we do not yet have.
```

Comment:
Link to bugs on NuGet for the desired functionality so we can delete this?

Reply:
Done. https://github.com/NuGet/Home/issues/5919